compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/cupertino/CupertinoOverscrollEffect.kt

```diff
@@ -445,30 +441,34 @@ class CupertinoOverscrollEffect(
 
 private class CupertinoOverscrollNode(
     val offset: Density.() -> IntOffset,
-    val coordinates: (LayoutCoordinates) -> Unit,
+    val onNodeRemeasured: (IntSize) -> Unit,
     val applyClip: Boolean
-): LayoutAwareModifierNode, DrawModifierNode, Modifier.Node() {
-
-    override val shouldAutoInvalidate: Boolean = true
+): LayoutModifierNode, LayoutAwareModifierNode, DrawModifierNode, Modifier.Node() {
+    override fun onRemeasured(size: IntSize) = onNodeRemeasured(size)
 
     override fun ContentDrawScope.draw() {
         if (applyClip) {
-            clipRect { this@draw.drawContentWithOffset() }
+            val offset = offset()
+            clipRect(
+                top = maxOf(0f, -offset.y.toFloat()),
+                bottom = size.height - maxOf(0, offset.y),
+                left = maxOf(0f, -offset.x.toFloat()),
+                right = size.width - maxOf(0, offset.x)
+            ) { this@draw.drawContent() }
```

Comment:
It took some time to understand this. Let's simplify to
```suggestion
            val bounds = Rect(offset().toOffset(), size)
            clipRect(size.toRect().intersect(bounds)) { this@draw.drawContent() }
```

Reply:
slightly different, but I got you point.